TX/Mavlink/Airside_Functions.cpp

```diff
@@ -0,0 +1,172 @@
+/**************************************************************************************************/
+// Author: Jingting Liu
+// April 9th, 2021
+
+// Groundside Encoding Modifications
+// Author: Bassel Al Omari
+// June 7th, 2021
+
+// to run the test without cmake:
+// gcc -g Airside_Functions.cpp
+// a.out
+
+// to run the test with cmake:
+// mkdir build, cd build
+// cmake ..
+// make
+// FUNCTIONS (executing the generated target, which is FUNCTIONS in this case)
+
+// The encoder and decoders are implemented based on Mavlink2 (not Mavlink1)
+// refer to this page for the stucture of mavlink messages
+// https://mavlink.io/en/guide/serialization.html
+
+/**************************************************************************************************/
+
```

Comment:
You should not be copying files rom other repos into yours. This causes many problems such as updates not being carried forward. 

Instead, you should do this:
1. Clone the Airside Mavlink repo in a separate project directory that shares the same parent directory as ground station
2. Import airside as a library in your code. For an example of how this is done, look at the CMakeLists.txt file in ./ZeroPilot-SW/Autopilot. 

Tbh, with comp around the corner I dont know how much time you have to implement this, but if you do have time I'd highly recommend doing it because I have made changes to the airside files...

Reply:
```txt
# Libraries
  add_subdirectory(boardfiles/Drivers/STM32F7xx_HAL_Driver)
  add_subdirectory(boardfiles/Middlewares/Third_Party/FreeRTOS)
  add_subdirectory(DataStructures)
  add_subdirectory(../../Mavlink2 build)

  file(GLOB_RECURSE C_SOURCES ../Common/Src/*.c ../Common/Src/stm32/*.c "Src/*.c" AttitudeManager/Src/*.c boardfiles/Src/*.c CControl/Src/*.c)
file(GLOB_RECURSE CXX_SOURCES ../Common/Src/*.cpp ../Common/Src/stm32/*.cpp "Src/*.cpp" AttitudeManager/Src/*.cpp PathManager/Src/*.cpp TelemetryManager/Src/*.cpp SensorFusion/Src/*.cpp)
  if(NOT NUCLEO)
    set(STARTUP_ASM startup_stm32f765xx.s)
    set(LINKER_SCRIPT ${PROJECT_SOURCE_DIR}/STM32F765ZGTx_FLASH.ld)
  else(NOT NUCLEO)
    set(STARTUP_ASM startup_stm32f767xx.s)
    set(LINKER_SCRIPT ${PROJECT_SOURCE_DIR}/STM32F767ZITx_FLASH.ld)
  endif(NOT NUCLEO)


  add_executable(${ELF_NAME} ${C_SOURCES} ${CXX_SOURCES} ${STARTUP_ASM})

  target_link_libraries(${ELF_NAME} STM32_HAL FreeRTOS DataStructures arm_cortexM7lfdp_math Mavlink_Airside)
```

Code that imports the library in the CMakeLists.txt

Reply:
Also look at the CMakeLists.txt in the airside telemetry repo. (I think it's in a PR right now). That shows how I made the repo into a library.